compiler/rustc_abi/src/lib.rs

```diff
@@ -346,6 +354,31 @@ impl TargetDataLayout {
         }
     }
 
+    /// Returns the theoretical maximum address.
+    ///
+    /// Note that this doesn't take into account target-specific limitations.
+    #[inline]
+    pub fn max_address(&self) -> u64 {
```

Comment:
This is a duplicate of `target_usize_max`. Not sure if there is a good way to share that code, though.

Reply:
I guess `target_usize_max` could directly forward to `max_address` (which could also be renamed to `target_usize_max`)?

Reply:
Miri has `target_{isize,usize}_{max,min}`. So if we do that we'd probably just want to move all of them to the ABI crate.